### PR TITLE
Add docs about how to add a new feature proposal in consensus-specs and online viewer support

### PR DESCRIPTION
**Pull request by hwwhww**
- Add `docs` folder
    - Add  `docs/docs/new-feature.md` doc as the new feature guide
- Add new online viewer support with `mkdocs`
    - The config is  `mkdocs.yml`
    - Since we don't want to change the `specs/*` path, I created a `copy_docs` makefile command to copy the specs files to `docs/` folder. The GitHub Actions job would call `make copy_docs` and build the docs viewer website.
    - demo: https://hwwhww.github.io/consensus-specs/
- Add  `fork_choice/.pages` and `specs/.pages`  files for `mkdocs-awesome-pages-plugin` navigation.
     - These files are for custom navigation orders or extra links.
- Fix some wrong links
- TODOs
    - [ ] Some links to files in `tests/` folder are still wrong. The simplest solution is using absolute links here. Anothor solution is to arrange the `tests` docs all together.
        ```
        WARNING  -  Documentation file 'README.md' contains a link to 'tests/formats/README.md' which is not found in the documentation files.
        WARNING  -  Documentation file 'README.md' contains a link to 'tests/generators/README.md' which is not found in the documentation files.
        WARNING  -  Documentation file 'README.md' contains a link to 'tests/core/pyspec/README.md' which is not found in the documentation files.
        WARNING  -  Documentation file 'specs/phase0/deposit-contract.md' contains a link to 'solidity_deposit_contract/deposit_contract.sol' which
                    is not found in the documentation files.
        ```
    - [ ] GitHub repo links only show the formal "release" instead of "pre-release" (release candidates). To avoid confusing, we should probably remove the versions or find another plug-in.
        image

### CODE REVIEW

File: docs/new-feature.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# How to add a new feature proposal in consensus-specs
+
+1. Create a folder under `./specs/features`.
```
Comment (djrtwo) [inline docs/new-feature.md:3]:
```suggestion
1. Create a folder under `./specs/_features`.
```

**Review by djrtwo — APPROVED** [review]
looks good! minor nit

**Comment by hwwhww** [comment]
I'm trying to use [mkdocs](https://www.mkdocs.org/) to build the docs website. Preview website: https://hwwhww.github.io/consensus-specs/

- Since we don't want to change the `specs/*` path, I created a `copy_docs` makefile command to copy the specs files to `docs/` folder. The GitHub Actions job would call `make copy_docs` and build the docs viewer website.
- The extra docs can also be stored in `docs` folder.
- We can add a light client specs index markdown file. I hope this can help address the issue that @etan-status [raised](https://github.com/ethereum/consensus-specs/pull/3259#pullrequestreview-1305195340)

@djrtwo @ralexstokes what do you think about the CI-built spec viewer website proposal?

**Comment by ralexstokes** [comment]
spec viewer website looks cool! I don't immediately see what it buys us over the default GitHub UI but it likely doesn't hurt anything to add it

**Comment by hwwhww** [comment]
@etan-status

I think it's doable, but I may need to write a custom `mkdocs` plug-in/extension for navigation. Noted in #3325 for now.

File: specs/_features/eip6110/beacon-chain.md
Comment (Md0135) [inline specs/_features/eip6110/beacon-chain.md:1]: Türkiye

File: specs/_features/sharding/p2p-interface.md
Comment (Md0135) [inline specs/_features/sharding/p2p-interface.md:1]: Türkiye

**Pull request merged**